admin/includes/classes/AdminRequestSanitizer.php

```diff
@@ -560,10 +670,13 @@ private function filterProductNameDeepRegex($parameterName)
     {
         $prodNameRegex = '~<\/?scri|on(load|mouse|error|read|key)(up|down)? ?=|[^(class|style)] ?= ?(\(|")|<!~i';
         if (isset($_POST[$parameterName])) {
+            // Add the parameterName to the base arrayname.
+            $this->arrayName = $this->setCurrentArrayName($parameterName, $this->arrayName);
             $this->debugMessages[] = 'PROCESSING PRODUCT_NAME_DEEP_REGEX == ' . $parameterName;
             foreach ($_POST[$parameterName] as $pKey => $pValue) {
+                $currentArrayName = $this->arrayName . '[' . $pKey . ']';
```

Comment:
For some reason I think I did this on purpose; however, because I don't know and didn't document the specific purpose, it seems like this should be rewritten to use the above `$this->setCurrentArrayName` function to assign the value back to `$currentArrayName`.